### ISSUE TITLE
[Deployment] Create a script to remove all CiviForm resources with a particular prefix

### ISSUE THREAD

**Issue by dkatzz** [civiform/civiform#6417]
#### Description
When going through the AWS deployment with Charlotte, we ran `bin/setup` and then had to start over, but then when we re-ran the command, we were getting errors that resources already existed. We tried running `bin/run` `destroy`, but that didn't seem to remove everything. Also, aws-nuke removes all resources from an account, so it only would work if a civic entity has separate organizations setup for each deployment and it requires an account alias (which charlotte didn't have).

We should create another command (or use the same destroy command) to clean up all resources created by CiviForm with a particular prefix.

<img width="1488" alt="Screenshot 2024-01-19 at 4 17 16 PM" src="https://github.com/civiform/civiform/assets/86739416/ea088ab4-8776-437e-9971-b17c9c244769">

#### What priority is this?

If we work on https://github.com/civiform/civiform/issues/6418, this becomes lower priority. But I'd probably advocate for doing this one first over the other. 

#### Associated civic entity

This came up when going through the charlotte deployment

#### Done when:

What conditions need to be met before this issue can be closed?

**Comment by rockycodes** [comment]
Update: We discovered through testing that it is sufficient to run `bin/destroy` before running `bin/setup` again. The trick is that when you run `bin/setup` twice in a row it asks if you would like to "delete backend state resources" which destroys the terraform tracking files and makes it so that terraform loses track of the files it should destroy with `bin/destroy`. The solution we settled on is to automatically run `terraform destroy` before deleting the  backend state resources if the user answers "yes" to that question when re-running `bin/setup`. See [this Slack thread](https://civiform.slack.com/archives/C04QU24UJSF/p1709576782146449) for more context.
